orchagent/portsorch.cpp

```diff
@@ -1239,14 +1241,71 @@ bool PortsOrch::setHostIntfsOperStatus(const Port& port, bool isUp) const
     return true;
 }
 
+void PortsOrch::updateDbPortFlapCounter(const Port& port, vector<FieldValueTuple>& old_tuples, vector<FieldValueTuple>& new_tuples) const
+{
+  SWSS_LOG_ENTER();
+
+  string flap_value;
+  bool check = false;
+
+  /* Fetching current flap counters from redis DB and incrementing it by 1 under any UP or DOWN event */
+
+  for (auto const &i : old_tuples)
+  {
+    if (fvField(i) == "flap_counter")
+    {
+      flap_value = fvValue(i);
+      check = true;
+      break;
+    }
+  }
+
+  if (!check)
+  {
+    return;
+  }
+
+  long flap_val = stol(flap_value);
+  flap_val = flap_val + 1;
+  string flaps;
+  stringstream flap_stream;
+  flap_stream << flap_val;
+  flaps = flap_stream.str();
+  FieldValueTuple flap_tuple("flap_counter", flaps);
+  new_tuples.push_back(flap_tuple);
+}
+
+void PortsOrch::updateDbPortLastFlapTime(vector<FieldValueTuple>& new_tuples) const
+{
+  SWSS_LOG_ENTER();
+
+  time_t now = time(0);
+  string date = ctime(&now);
+
+  // convert now to tm struct for UTC
+  tm *gmtm = gmtime(&now);
+  date = asctime(gmtm);
+  FieldValueTuple tuple("last_flap", date);
+  new_tuples.push_back(tuple);
+}
+
 void PortsOrch::updateDbPortOperStatus(const Port& port, sai_port_oper_status_t status) const
 {
     SWSS_LOG_ENTER();
 
-    vector<FieldValueTuple> tuples;
+    vector<FieldValueTuple> old_tuples;
+    vector<FieldValueTuple> new_tuples;
+
+    bool exist = m_portTable->get(port.m_alias, old_tuples);
+    if (!exist)
+    {
+      return;
+    }
+    updateDbPortLastFlapTime(new_tuples);
```

Comment:
move these two functions into `updatePortOperStatus` function instead of here.
under the two lines:
```
updateDbPortOperStatus(port, status);
port.m_oper_status = status;
```